src/lib/components/layout/Sidebar.svelte

```diff
@@ -49,6 +52,12 @@
 	let showDropdown = false;
 
 	let filteredChatList = [];
+	let paginationScrollThreashold = 0.6;
+	let nextPageLoading = false;
+	let tagView = false;
+	let chatPagniationComplete = false;
+
+	pageLimit.set(20);
```

Comment:
could cause issues on large screens, or zoomed out. 

only 20 chats load, so if the list of chats is smaller than the screen height, there is no scroll event that can be made to then get the next 

maybe get screen height from the window and determine the pageLimit from that. Could also improve performance on mobile, only a few chats need to be loaded in comparison to a larger screen.